src/routes/safe/components/Apps/components/SafeAppsErrorBoundary.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, ErrorInfo } from 'react'
+import styled from 'styled-components'
+import { Text, Link, Icon, FixedIcon, Title } from '@gnosis.pm/safe-react-components'
+import { RouteComponentProps, withRouter } from 'react-router-dom'
+import { extractPrefixedSafeAddress, generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+
+type SafeAppsErrorBoundaryProps = {
+  children?: ReactNode
+} & RouteComponentProps
+
+type SafeAppsErrorBoundaryState = {
+  hasError: boolean
+  error?: Error
+}
+
+class SafeAppsErrorBoundary extends React.Component<SafeAppsErrorBoundaryProps, SafeAppsErrorBoundaryState> {
+  public state: SafeAppsErrorBoundaryState = {
+    hasError: false,
+  }
+
+  constructor(props: SafeAppsErrorBoundaryProps) {
+    super(props)
+    this.state = { hasError: false }
+
+    this.handleGoBack = this.handleGoBack.bind(this)
+  }
+
+  public componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
+    console.error('Uncaught error:', error, errorInfo)
+  }
+
+  public static getDerivedStateFromError(error: Error): SafeAppsErrorBoundaryState {
+    return { hasError: true, error }
+  }
+
+  private handleGoBack(): void {
+    this.props.history.push(generateSafeRoute(SAFE_ROUTES.APPS, extractPrefixedSafeAddress()))
+  }
+
+  public render(): React.ReactNode {
+    if (this.state.hasError) {
+      return (
+        <Wrapper>
+          <Content>
+            <Title size="md">Safe App cold not be loaded.</Title>
+            <FixedIcon type="networkError" />
+
+            <div>
+              <Text size="xl" as="span">
+                In case the problem persists, please reach out to us via{' '}
+              </Text>
+              <LinkWrapper>
+                <a target="_blank" href="https://chat.gnosis-safe.io" rel="noopener noreferrer">
+                  <Text color="primary" size="lg" as="span">
+                    Discord
+                  </Text>
+                </a>
+                <Icon type="externalLink" color="primary" size="sm" />
+              </LinkWrapper>
+            </div>
```

Comment:
This looks like the same as our current `ErrorBoundary`. Probably worthwhile extracting it into it's own component. What do you think?

Reply:
Sounds good to me!